crates/wadm/src/scaler/spreadscaler/mod.rs

```diff
@@ -266,6 +276,16 @@ impl<S: ReadStore + Send + Sync + Clone> Scaler for ComponentSpreadScaler<S> {
             .collect::<Vec<Command>>();
         trace!(?commands, "Calculated commands for component scaler");
 
+        // Detect spread requirement violations
+        if let Some(message) = detect_spread_requirement_violations(
+            &self.spread_requirements,
+            &hosts,
+            &component_instances_per_eligible_host,
+            &commands,
+        ) {
+            spread_status.push(StatusInfo::failed(&message));
+        }
```

Comment:
In this case, where there are conflicting spreads, doesn't that mean that the scalers will reconcile forever trying to adjust the number of components on a host? I might be misunderstanding, but if that's the case, what do you think about returning no commands here in order to ensure that we don't deadloop?